.pre-commit-config.yaml

```diff
@@ -0,0 +1,17 @@
+repos:
+  - repo: https://gitlab.com/pycqa/flake8
+    rev: 3.9.2
+    hooks:
+      - id: flake8
+        args: [--config=setup.cfg]
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+      - id: no-commit-to-branch
+      - id: check-added-large-files
+      - id: check-docstring-first
+      - id: check-toml
+      - id: check-yaml
+        exclude: packaging/.*
+      - id: check-json
```

Comment:
These are basically smoke tests if these files have a correct syntax. We need to exclude some `.yaml`'s in `packaging/*` since they are templates and thus fail this test:

https://github.com/pytorch/vision/blob/12fd3a625a044a454cca3dbb2187e78efe1b4596/packaging/torchvision/meta.yaml#L1-L6

Reply:
Apart from `./gallery/assets/imagenet_class_index.json` I don't think there are `.json` files that aren't automatically generated by other stuff, so maybe we can remove this one?

Reply:
I see this was resolved, did you decide to keep the json hook after all?

Reply:
I just failed to push my local commit. Same is true for the `no-commit-to-branch` and `check-added-large-files` hooks.